### ISSUE TITLE
Unhandled Exception: Null check operator used on a null value

### ISSUE THREAD

**Issue by spk22** [parse-community/Parse-SDK-Flutter#646]
I'm using _parse_server_sdk_flutter: ^3.0.0_ in my flutter app.
When I try to register a user, this error keeps recurring, although the user does get registered on the server (as visible from Back4App dashboard).

> E/flutter (21228): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: Null check operator used on a null value

> E/flutter (21228): #0 ParseUser._handleResponse

On clicking this error, I'm taken to the function static Future _handleResponse located in parse_user.dart file.  

```
if (responseData.containsKey(keyVarObjectId)) {
      user.sessionToken = responseData[keyParamSessionToken];
      ParseCoreData().setSessionId(user.sessionToken!);
    }
```

I think the problem is with the exclamation sign (!) used at the end of sessionToken. Since the setSessionId() method takes a non-nullable String, passing a nullable value (user.sessionToken!) is not permissible in null-safe Dart code.

How to mitigate this error? Plz help.

**Comment by RodrigoSMarques** [comment]
The fix was made in the code and it's here #638 

This error occurs if you have the user email verification option enabled.

If you disable it, you can login.

We will release a new version next week.

**Comment by spk22** [comment]
Thanks, I disabled the email verification and it didn't reproduce. 
Eagerly waiting for the next release. Good luck!

Please consider this issue closed.
